samcli/commands/local/lib/sam_api_provider.py

```diff
@@ -77,17 +80,51 @@ def _extract_from_serverless_api(self, logical_id, api_resource, collector, cwd=
         body = properties.get("DefinitionBody")
         uri = properties.get("DefinitionUri")
         binary_media = properties.get("BinaryMediaTypes", [])
+        cors = self._extract_cors(properties.get("Cors", {}))
         stage_name = properties.get("StageName")
         stage_variables = properties.get("Variables")
-
         if not body and not uri:
             # Swagger is not found anywhere.
             LOG.debug("Skipping resource '%s'. Swagger document not found in DefinitionBody and DefinitionUri",
                       logical_id)
             return
+
         self.extract_swagger_route(logical_id, body, uri, binary_media, collector, cwd=cwd)
         collector.stage_name = stage_name
         collector.stage_variables = stage_variables
+        collector.cors = cors
+
+    def _extract_cors(self, cors_prop):
+        """
+        Extract Cors property from AWS::Serverless::Api resource by reading and parsing Swagger documents. The result
+        is added to the Api.
+
+        Parameters
+        ----------
+        cors_prop : dict
+            Resource properties for Cors
+        """
+        cors = None
+        if cors_prop and isinstance(cors_prop, dict):
+            allow_methods = cors_prop.get("AllowMethods", ','.join(Route.ANY_HTTP_METHODS))
+
+            if allow_methods and "OPTIONS" not in allow_methods and "options" not in allow_methods:
```

Comment:
You need to normalize `allow_methods` here, just comparing OPTIONS and options is not enough, unless SAM or API gateway only allow these two ways for the strings.